Makefile

```diff
@@ -0,0 +1,39 @@
+.PHONY: build clean test help default lint
+
+PROJECT=github.com/praekeltfoundation/vault-plugin-auth-mesos
+
+BIN_NAME=vault-plugin-auth-mesos
+
+VERSION := $(shell grep "const Version " version/version.go | sed -E 's/.*"(.+)"$$/\1/')
```

Comment:
Reason why this is ` := ` and others are not?

Reply:
No idea. It came from the cookiecutter thing's Makefile and I haven't looked up what the syntax actually means.

I'll do a pass through the Makefile cleaning up and understanding everything at some point, probably before merging this PR.

Reply:
Paraphrased from the GNU make docs:
A variable set using `=` is "recursively expanded" -- any substitutions or whatever are evaluated when the variable is referenced.
A variable set using `:=` (or `::=`) is "simply expanded" -- everything is evaluated when the variable is set.
More detail at https://www.gnu.org/software/make/manual/html_node/Flavors.html